src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -75,6 +75,17 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
 
   override def settings = currentSettings
 
+  private[this] var currentReporter: Reporter = { reporter = reporter0 ; currentReporter }
+
+  def reporter: Reporter = currentReporter
+  def reporter_=(newReporter: Reporter): Unit =
+    currentReporter = newReporter match {
+      case _: reporters.ConsoleReporter | _: reporters.LimitingReporter => newReporter
+      case _ if settings.maxerrs.isSetByUser && settings.maxerrs.value < settings.maxerrs.default =>
```

Comment:
why do we check `settings.maxerrs.value < settings.maxerrs.default` here?

Reply:
That is a back door. Instead of a sentinel value like maxerrs=-1, if they're explicitly asking for a "huge" maxerrs, then trust their reporter to do whatever filtering and limiting it wants. Only if the user asks to "restrict" the maxerrs do we take this extraordinary step of intervening.

Probably this will never happen IRL, but for example: a reporter that removes similar errors might be trusted to respect maxerrs. Put another way, this change is really because sbt reporter doesn't respect maxerrs and when I set it to 5, I want that to work and not have to look in the sbt docs for their special setting.

Reply:
OK, doppelt genäht hält besser :-)

Reply:
💯